packages/sdk/src/protocols/base_protocol.ts

```diff
@@ -1,15 +1,206 @@
+import type { Peer, PeerId } from "@libp2p/interface";
+import { ConnectionManager } from "@waku/core";
+import { BaseProtocol } from "@waku/core/lib/base_protocol";
 import { IBaseProtocolSDK } from "@waku/interfaces";
+import { Logger } from "@waku/utils";
 
 interface Options {
   numPeersToUse?: number;
+  maintainPeersInterval?: number;
 }
 
 const DEFAULT_NUM_PEERS_TO_USE = 3;
+const DEFAULT_MAINTAIN_PEERS_INTERVAL = 30_000;
 
 export class BaseProtocolSDK implements IBaseProtocolSDK {
-  public readonly numPeers: number;
+  public readonly numPeersToUse: number;
+  public peers: Peer[] = [];
+  private maintainPeersIntervalId: NodeJS.Timeout | undefined;
```

Comment:
it should be a number 
more info https://github.com/waku-org/js-waku/pull/1970#discussion_r1584551030

Reply:
`window.setInterval` should be fine, as it is just typing issue